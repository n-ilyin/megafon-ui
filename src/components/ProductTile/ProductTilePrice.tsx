import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTilePrice.less';
import AnimationValue from './ProductTileValue';
import { IPayment } from './types';
import { productTilePriceProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-price');
class ProductTilePrice extends React.Component<IPayment> {
    static propTypes = productTilePriceProps;

    render() {
        const { title, value, unitExtra, unitValue, discount } = this.props;
        const valueElem = <AnimationValue value={(discount || value) + ` ${unitValue}`} />;
        const unitElem = <span className={cn('payment-period')}>{unitExtra}</span>;

        return (
            <div className={cn('', { discount: !!discount })}>
                <div className={cn('discount-condition')}>{title}</div>
                <span className={cn('old-price')}>
                    <span className={cn('old-price-value')}>
                        {discount ? `${value} ${unitValue}` : ' '}
                    </span>
                </span>
                <span className={cn('actual-price')}>
                    {valueElem} {unitElem}
                </span>
            </div>
        );
    }
}

export default ProductTilePrice;
