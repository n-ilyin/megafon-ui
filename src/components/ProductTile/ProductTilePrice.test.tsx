import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTilePrice from './ProductTilePrice';

const props = {
    title: 'Новым абонентам 1000 ₽ на счёт',
    value: '1500',
    unitExtra: '₽',
    unitValue: 'за 30 дней',
};

describe('<ProductTilePrice />', () => {
    it('renders ProductTilePrice', () => {
        const wrapper = shallow(<ProductTilePrice {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTilePrice with discount', () => {
        const wrapper = shallow(<ProductTilePrice {...props} discount="1000" />);

        expect(wrapper).toMatchSnapshot();
    });
});
