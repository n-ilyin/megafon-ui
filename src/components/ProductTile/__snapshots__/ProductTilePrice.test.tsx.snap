// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTilePrice /> renders ProductTilePrice 1`] = `
<div
  className="mfui-product-tile-price"
>
  <div
    className="mfui-product-tile-price__discount-condition"
  >
    Новым абонентам 1000 ₽ на счёт
  </div>
  <span
    className="mfui-product-tile-price__old-price"
  >
    <span
      className="mfui-product-tile-price__old-price-value"
    >
       
    </span>
  </span>
  <span
    className="mfui-product-tile-price__actual-price"
  >
    <ProductTileValue
      value="1500 за 30 дней"
    />
     
    <span
      className="mfui-product-tile-price__payment-period"
    >
      ₽
    </span>
  </span>
</div>
`;

exports[`<ProductTilePrice /> renders ProductTilePrice with discount 1`] = `
<div
  className="mfui-product-tile-price mfui-product-tile-price_discount"
>
  <div
    className="mfui-product-tile-price__discount-condition"
  >
    Новым абонентам 1000 ₽ на счёт
  </div>
  <span
    className="mfui-product-tile-price__old-price"
  >
    <span
      className="mfui-product-tile-price__old-price-value"
    >
      1500 за 30 дней
    </span>
  </span>
  <span
    className="mfui-product-tile-price__actual-price"
  >
    <ProductTileValue
      value="1000 за 30 дней"
    />
     
    <span
      className="mfui-product-tile-price__payment-period"
    >
      ₽
    </span>
  </span>
</div>
`;
